src/hooks/useSmartSearch.ts

```diff
@@ -1,19 +1,27 @@
 import { useState } from 'react';
-import { SmartSearchFilterWithId, ZetkinSmartSearchFilter } from 'types/smartSearch';
+import { FILTER_TYPE, OPERATION, SmartSearchFilterWithId, ZetkinSmartSearchFilter } from 'types/smartSearch';
 
 type InitialFilters = ZetkinSmartSearchFilter[]
 
 type UseSmartSearch = {
     addFilter: (filter: ZetkinSmartSearchFilter) => void; // addSmartSearchFilter
+    addFilterToStart: (filter: ZetkinSmartSearchFilter) => void; // add All Filter to start
     deleteFilter: (id: number) => void; // removeSmartSearchFilter
     editFilter: (id: number, newFilterValue: SmartSearchFilterWithId) => void; // editSmartSearchFilter
     filters: ZetkinSmartSearchFilter[];
     filtersWithIds: SmartSearchFilterWithId[];
 }
 
 const useSmartSearch = (initialFilters: InitialFilters = []): UseSmartSearch => {
-    const initialFiltersWithIds = initialFilters.map((filter, index) => ({ ...filter, id: index }));
-    const [filtersWithIds, setFiltersWithIds] = useState<SmartSearchFilterWithId[]>(initialFiltersWithIds);
+
+    // correctly configure legacy queries to only have the All filter in the first position with op: 'add'
+    const indexOfAllFilter = initialFilters.findIndex(f => f.type === FILTER_TYPE.ALL);
+    const normalizedFiltersWithIds = initialFilters
+        .filter((filter, index) => index > indexOfAllFilter ||
+        (index === indexOfAllFilter && filter.op === OPERATION.ADD))
+        .map((filter, index) => ({ ...filter, id: index }));
```

Comment:
Correct me if I'm wrong but I don't believe this will catch a scenario where there are two `all` filters. I think you would have to find the index of _the last_ `all` filter rather than the first one, as here.